gnovm/pkg/gnolang/realm.go

```diff
@@ -669,6 +688,37 @@ func (rlm *Realm) markDirtyAncestors(store Store) {
 	}
 }
 
+// clearDeleted removes any objects from .updated, .created,
+// and .newEscaped slices that are also present in the .deleted slice.
+func (rlm *Realm) clearDeleted() {
```

Comment:
I'm still unsure about this, but I would'nt know any better, so I'll let jae comment if he thinks this should b echanged.

Reply:
yeah agree we should be careful to modify any realm logic.

Reply:
This definitely shouldn't happen.
We use slices for performance, basically append-only operation.
It's better to iterate over some things that no longer need to be iterated over,
than to construct a map and filter.

Reply:
yeah construct a map is not good.

aside, do you mean we should not have `clearDeleted`?

it's not only about iterating things that no longer need to be iterated over, but also fails some sanity check, e.g. `ensureUniq(rlm.created, rlm.updated, rlm.deleted)`

Reply:
so not deleted them, but to modify the downstream code, right?

Reply:
updated.